### PR TITLE
Upgrade Compose and Kotlin to latest version

### PR DESCRIPTION
**Pull request by zach-klippenstein**

### CODE REVIEW

File: .buildscript/configure-compose.gradle [hunk 1]
```diff
@@ -19,7 +19,7 @@ android {
     compose true
   }
   composeOptions {
-    kotlinCompilerVersion "1.4.0-dev-withExperimentalGoogleExtensions-20200720"
```
Comment (patrickyin) [inline .buildscript/configure-compose.gradle:22]: Like this change!

File: compose-tooling/src/main/java/com/squareup/workflow/ui/compose/tooling/PlaceholderViewFactory.kt [hunk 2]
```diff
@@ -17,28 +17,28 @@
...
-import androidx.ui.graphics.Paint
-import androidx.ui.graphics.Shadow
-import androidx.ui.graphics.drawscope.DrawScope
-import androidx.ui.graphics.drawscope.drawCanvas
-import androidx.ui.graphics.drawscope.rotate
-import androidx.ui.graphics.withSaveLayer
-import androidx.ui.layout.fillMaxSize
-import androidx.ui.text.TextStyle
-import androidx.ui.text.style.TextAlign
+import androidx.compose.foundation.Box
```
Comment (patrickyin) [inline compose-tooling/src/main/java/com/squareup/workflow/ui/compose/tooling/PlaceholderViewFactory.kt:20]: Google's renaming is going well. hope they can rename all of the packages soon

File: core-compose/src/androidTest/java/com/squareup/workflow/ui/compose/RenderAsStateTest.kt [hunk 3]
```diff
@@ -41,6 +40,7 @@ import okio.ByteString.Companion.decodeBase64
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
+import androidx.ui.test.waitForIdle as waitForIdle1
```
Comment (patrickyin) [inline core-compose/src/androidTest/java/com/squareup/workflow/ui/compose/RenderAsStateTest.kt:43]: Is there any clashes if we use `waitForIdle`?

File: samples/src/main/java/com/squareup/sample/launcher/SampleLauncherApp.kt [hunk 4]
```diff
@@ -18,51 +18,51 @@ package com.squareup.sample.launcher
...
-import androidx.ui.unit.PxBounds
-import androidx.ui.unit.dp
-import androidx.ui.unit.height
-import androidx.ui.unit.width
-import com.squareup.sample.R
+import com.squareup.sample.R.string
 
 @Composable fun SampleLauncherApp() {
-  MaterialTheme(colors = darkColorPalette()) {
+  MaterialTheme(colors = darkColors()) {
```
Comment (patrickyin) [inline samples/src/main/java/com/squareup/sample/launcher/SampleLauncherApp.kt:57]: We only use the dark colors in this `SampleLauncherApp`? Or we want to determine the system settings by using `isSystemInDarkTheme()`?

File: .buildscript/configure-compose.gradle (hunk 1 above)
Comment (zach-klippenstein) [reply .buildscript/configure-compose.gradle:22]: Oops, according to Google devs this is still required, albeit with a different value. But this compiled, so not sure why.

File: core-compose/src/androidTest/java/com/squareup/workflow/ui/compose/RenderAsStateTest.kt (hunk 3 above)
Comment (zach-klippenstein) [reply core-compose/src/androidTest/java/com/squareup/workflow/ui/compose/RenderAsStateTest.kt:43]: Oops, I think this was a fat-finger.

File: samples/src/main/java/com/squareup/sample/launcher/SampleLauncherApp.kt (hunk 4 above)
Comment (zach-klippenstein) [reply samples/src/main/java/com/squareup/sample/launcher/SampleLauncherApp.kt:57]: Only dark. This sample isn't intended to showcase reactive theming.

**Pull request merged**